frontend/src/lib/components/CommandPalette/commandPaletteLogic.ts

```diff
@@ -0,0 +1,681 @@
+            0,
+            {
+                setInput: () => 0,
+                executeResult: () => 0,
+                activateFlow: () => 0,
+                onArrowUp: (previousIndex) => (previousIndex > 0 ? previousIndex - 1 : 0),
+                onArrowDown: (previousIndex, { maxIndex }) => (previousIndex < maxIndex ? previousIndex + 1 : maxIndex),
+            },
+        ],
+        hoverResultIndex: [
+            null as number | null,
+            {
+                onMouseEnterResult: (_, { index }) => index,
+                onMouseLeaveResult: () => null,
+                onArrowUp: () => null,
+                onArrowDown: () => null,
+                activateFlow: () => null,
+            },
+        ],
+        input: [
+            '',
+            {
+                setInput: (_, { input }) => input,
+                activateFlow: () => '',
+                executeResult: () => '',
+            },
+        ],
+        activeFlow: [
+            null as CommandFlow | null,
+            {
+                activateFlow: (_, { flow }) => flow,
+            },
+        ],
+        rawCommandRegistrations: [
+            {} as CommandRegistrations,
+            {
+                registerCommand: (commands, { command }) => {
+                    return { ...commands, [command.key]: command }
+                },
+                deregisterCommand: (commands, { commandKey }) => {
+                    const cleanedCommands = { ...commands }
+                    delete cleanedCommands[commandKey]
+                    return cleanedCommands
+                },
+            },
+        ],
+    },
+
+    listeners: ({ actions, values }) => ({
+        showPalette: () => {
+            window.posthog?.capture('palette shown')
+        },
+        togglePalette: () => {
+            if (values.isPaletteShown) window.posthog?.capture('palette shown')
+        },
+        executeResult: ({ result }: { result: CommandResult }) => {
+            if (result.executor === true) {
+                actions.activateFlow(null)
+            } else {
+                const possibleFlow = result.executor?.() ?? null
+                actions.activateFlow(possibleFlow)
+                if (!possibleFlow) actions.hidePalette()
+            }
+            // Capture command execution, without useless data
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            const { icon, index, ...cleanedResult } = result
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            const { resolver, ...cleanedCommand } = cleanedResult.source
+            cleanedResult.source = cleanedCommand
+            window.posthog?.capture('palette command executed', cleanedResult)
+        },
+        deregisterAllWithMatch: ({ keyPrefix }) => {
+            for (const command of Object.values(values.commandRegistrations)) {
+                if (command.key.includes(keyPrefix) || command.scope.includes(keyPrefix)) {
```

Comment:
could `includes` eventually yield false positives? shouldn't it be `startsWith` instead? Otherwise I'd remove the variable to something other than key**Prefix**

Reply:
It does. To be honest I'm not sure why this is the way it is, paging @EDsCODE?

Reply:
ah there was something smelly with this. I meant this deregister function to just target the scope directly and nothing else but when I was doing the initial refactor I ran through this action too quickly so the operator here should actually just be a strict equality on command.scope. If someone pushes the change, I'd suggest making sure all the deregister calls are related to scope now and don't have a stray dependence on the key

Reply:
Replaced `deregisterAllWithMatch` with `deregisterScope`.